lib/repl.js

```diff
@@ -282,17 +282,25 @@ function REPLServer(prompt,
   self._domain.on('error', function debugDomainError(e) {
     debug('domain error');
     const top = replMap.get(self);
+    var out;
     internalUtil.decorateErrorStack(e);
-    if (e instanceof SyntaxError && e.stack) {
+
+    if (!((e instanceof Error) && typeof e.stack === 'string' &&
+        e.stack.length > 0)) {
```

Comment:
Just noting that whatever is done here would need to factor in compatibility with `internal/errors.js`